src/Domain/Creators/TimeDependentOptions/ShapeMap.cpp

```diff
@@ -130,9 +137,43 @@ FunctionsOfTimeMap get_shape_and_size(
     check_fot.template operator()<2>(shape_name);
     check_fot.template operator()<3>(size_name);
 
-    result[shape_name] =
+    // Not const in case we move it below
+    auto temporary_shape_fot =
         volume_fots.at(shape_name)
             ->create_at_time(initial_time, shape_expiration_time);
+    const auto shape_funcs =
+        temporary_shape_fot->func_and_2_derivs(initial_time);
+
+    const size_t file_l_max = -1 + sqrt(shape_funcs[0].size() / 2);
+
+    // Prolong or restrict if necessary, otherwise just use the exact function
+    // of time from the volume file
+    if (file_l_max != l_max) {
+      ModalVector shape_coefs{};
+      std::array<DataVector, 3> new_shape_funcs{};
+      for (size_t i = 0; i < shape_funcs.size(); i++) {
+        shape_coefs.set_data_ref(
+            const_cast<DataVector&>(gsl::at(shape_funcs, i)).data(),  // NOLINT
+            gsl::at(shape_funcs, i).size());
+        // The centers and radii of the following don't matter. Only their l_max
+        // and their coefficients
```

Comment:
Can you just create a YlmSpherepack object without the unused radius?

Reply:
Yeah, I had just copied other code. This new way is much cleaner.